tests/TestCase/ORM/Association/HasManyTest.php

```diff
@@ -770,6 +770,47 @@ public function testSubqueryWithLimitAndOrder(): void
         $this->assertCount(1, $result[1]->articles);
     }
 
+    /**
+     * Tests that eager loaded associations with limit and order properly
+     * propagate those clauses to the subquery strategy.
+     * This test addresses issue #11395
+     */
+    public function testSubqueryPreservesLimitAndOrderForEagerLoading(): void
+    {
+        $this->skipIf(ConnectionManager::get('test')->getDriver() instanceof Sqlserver, 'Sql Server does not support ORDER BY on field not in GROUP BY');
+
+        $authors = $this->getTableLocator()->get('Authors');
+        $authors->Articles->setStrategy(Association::STRATEGY_SUBQUERY);
+
+        // Test with both order and limit - they should be preserved together
+        $query = $authors->find()
+            ->contain(['Articles' => function ($q) {
+                return $q->orderBy(['Articles.title' => 'ASC'])
+                    ->limit(2);
+            }])
+            ->orderBy(['Authors.name' => 'DESC'])
+            ->limit(1);
+
+        $result = $query->first();
+
+        // Verify that we got a result
+        $this->assertNotNull($result);
+
+        // If the author has articles, verify they are properly limited and ordered
+        if (!empty($result->articles)) {
+            // Check that we got at most 2 articles due to the limit
+            $this->assertLessThanOrEqual(2, count($result->articles));
```

Comment:
Why would there be fewer? Couldn't this test silently fail by not running in the future if fixture rows were reduced?

Reply:
More explicit test data now.
Feel free to adjust, though.

Is the approach valid in general?